File: src/data/createSchedule.py
import sqlite3
import json
import sys

data = json.load(open(sys.argv[1]))

data = data['shifts']

days = 0
hours = 0
onyens = 0
insertQuery = '''INSERT INTO Shifts(day, hour, onyen) VALUES ('{}', '{}', '{}')'''

while days < 7:
        while hours < 24:
                while onyens < len(data[days][hours]['onyens']):
                        insertQuery.format(days, hours, data[days][hours]['onyens'][onyens])
                        print(days)
                        print(hours)
                        print(data[days][hours]['onyens'][onyens])
                        # print(insertQuery)
                        onyens = onyens + 1
                onyens = 0
                hours = hours + 1
        hours = 0
        days = days + 1

File: src/data/createSchedule.js
const MongoClient = require('mongodb').MongoClient;
const assert = require('assert');
var data = require('./currentSchedule.json');
data = data['shifts']

MongoClient.connect('mongodb://localhost:27017', (err, client) => {
  assert.equal(null, err);
  console.log('Connected successfully')
  const db = client.db('shiftoverflow');
  // Uncomment lines below when schedule is updated
  //deleteData(db);
  //insertData(db, data);
  client.close();
})

let insertData = (db, data) => {
  let schedule = db.collection('schedule');
  for (let days = 0; days < 7; days++) {
    for (let hours = 0; hours < 24; hours++) {
      for (let onyens = 0; onyens < data[days][hours]['onyens'].length; onyens++) {
        schedule.insertOne({
          day: days,
          hour: hours,
          onyen: data[days][hours]['onyens'][onyens] 
        });
        console.log(data[days][hours]['onyens'][onyens])
      }
    }
  }
}

let deleteData = (db) => {
  db.collection('schedule').deleteMany({});
}
